### PR TITLE
Prepare 0.18.0 release

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

To prepare for the upcoming 0.18.0 release this commit moves all the
release notes into a 0.18 subdirectory to keep them separate from future
development notes post-release. It also adds a release note with the
prelude section providing the high level overview of the release. When
this merges it should be the commit used to tag the 0.18.0 release.

### Details and comments

TODO:

- [x] Rewrite and fix release notes

### CODE REVIEW

File: releasenotes/notes/0.18/higher-order-spsa-f2eddb44b00b82e0.yaml
Comment (Cryoris) [inline releasenotes/notes/0.18/higher-order-spsa-f2eddb44b00b82e0.yaml:9]:
```suggestion
    algorithm, a generalization of the 2-SPSA algorithm, estimates the
    Quantum Fisher Information Matrix instead of a the Hessian to obtain a
```

File: releasenotes/notes/0.18/higher-order-spsa-f2eddb44b00b82e0.yaml
Comment (Cryoris) [inline releasenotes/notes/0.18/higher-order-spsa-f2eddb44b00b82e0.yaml:19]: Since we refer to 2-SPSA in the QNSPSA reno, it might be nice to use this abbreviation here too 🙂 
```suggestion
    Second order SPSA, or 2-SPSA, is an extension of the ordinary SPSA algorithm that
```

File: releasenotes/notes/0.18/prepare-0.18-9a0fb87005e314fd.yaml
Comment (ajavadia) [inline releasenotes/notes/0.18/prepare-0.18-9a0fb87005e314fd.yaml:6]:
```suggestion
    :class:`~qiskit.transpiler.passes.DynamicalDecoupling`, which when combined with
```

File: releasenotes/notes/0.18/prepare-0.18-9a0fb87005e314fd.yaml
Comment (ajavadia) [inline releasenotes/notes/0.18/prepare-0.18-9a0fb87005e314fd.yaml:10]: But the techniques are general and applicable to optimizing any circuits.

**Comment by mtreinish** [comment]
Ok, I've finished my first pass at the release notes we have so far. So this is basically ready for review now. But I'm leaving it as on hold until we're ready to release so that I updated all the release notes that for the open PRs for the release which haven't merged yet.

File: releasenotes/notes/0.18/add-alignment-management-passes-650b8172e1426a73.yaml
Comment (ajavadia) [inline releasenotes/notes/0.18/add-alignment-management-passes-650b8172e1426a73.yaml:11]:
```suggestion
    If a backend has a restriction on the alignment of
```

File: releasenotes/notes/0.18/add-alignment-management-passes-650b8172e1426a73.yaml
Comment (ajavadia) [inline releasenotes/notes/0.18/add-alignment-management-passes-650b8172e1426a73.yaml:12]:
```suggestion
    :class:`~qiskit.circuit.Measure` instructions (in terms of quantization in time), the
```

File: releasenotes/notes/0.18/add-alignment-management-passes-650b8172e1426a73.yaml
Comment (ajavadia) [inline releasenotes/notes/0.18/add-alignment-management-passes-650b8172e1426a73.yaml:30]:
```suggestion
    In the built-in :mod:`~qiskit.transpiler.preset_passmangers` used by the
    :func:`~qiskit.compiler.transpile` function, these passes get automatically
    triggered if the alignment constraint, either via the dedicated
    ``alignment`` kwarg on :func:`~qiskit.compiler.transpiler` or as an
    ``alignment`` attribute in the
    :class:`~qiskit.providers.models.BackendConfiguration` object of the
    backend being targetted, is set to > 1.
```

File: releasenotes/notes/0.18/add-alignment-management-passes-650b8172e1426a73.yaml
Comment (ajavadia) [inline releasenotes/notes/0.18/add-alignment-management-passes-650b8172e1426a73.yaml:40]: I think this paragraph is redundant given the above
```suggestion
     
     For example:
```

Comment (mtreinish) [reply releasenotes/notes/0.18/add-alignment-management-passes-650b8172e1426a73.yaml:40]: I included it because this was a separate release note. I split it into 2 notes/bullet points one for the new passes and one for the new `transpile()` option. But, yeah it probably make more sense as a single big note.

File: releasenotes/notes/0.18/add-alignment-management-passes-650b8172e1426a73.yaml
Comment (1ucian0) [inline releasenotes/notes/0.18/add-alignment-management-passes-650b8172e1426a73.yaml:7]:
```suggestion
    :mod:`qiskit.transpiler.passes` module. These passes are a hardware aware
```

File: releasenotes/notes/0.18/add-scipy-optimizer-d1b6e9b9dcb20356.yaml
Comment (1ucian0) [inline releasenotes/notes/0.18/add-scipy-optimizer-d1b6e9b9dcb20356.yaml:29]:
```suggestion
    optimizer class are deprecated and will be removed in a future release. Instead, you can just
```

File: releasenotes/notes/0.18/fix-opflow-coeff-bug-5557c50cc2f47401.yaml
Comment (1ucian0) [inline releasenotes/notes/0.18/fix-opflow-coeff-bug-5557c50cc2f47401.yaml:6]:
```suggestion
    This also fixed the calculations
```

File: releasenotes/notes/0.18/fix_initializer_global_phase-977eea9d416c9918.yaml
Comment (1ucian0) [inline releasenotes/notes/0.18/fix_initializer_global_phase-977eea9d416c9918.yaml:5]:
```suggestion
    instruction which was not correctly setting the global phase of the
```

File: releasenotes/notes/0.18/grad_wrapper_expectation-5214882ba482c62f.yaml
Comment (1ucian0) [inline releasenotes/notes/0.18/grad_wrapper_expectation-5214882ba482c62f.yaml:6]:
```suggestion
    not diagonal an incorrect result would be returned in some situations.
```

File: releasenotes/notes/0.18/higher-order-spsa-f2eddb44b00b82e0.yaml
Comment (1ucian0) [inline releasenotes/notes/0.18/higher-order-spsa-f2eddb44b00b82e0.yaml:9]:
```suggestion
    Quantum Fisher Information Matrix instead of the Hessian to obtain a
```

File: releasenotes/notes/0.18/latex_style_user_math_text-266f9b411317ae6b.yaml
Comment (1ucian0) [inline releasenotes/notes/0.18/latex_style_user_math_text-266f9b411317ae6b.yaml:20]:
```suggestion
    change a gate color::
```

File: releasenotes/notes/0.18/lo-freqs-circuit-c2ec161873eb4a68.yaml
Comment (1ucian0) [inline releasenotes/notes/0.18/lo-freqs-circuit-c2ec161873eb4a68.yaml:40]:
```suggestion
    be used. If the job level values are also missing, an error will be thrown. For example, if
```

File: releasenotes/notes/0.18/lo-freqs-circuit-c2ec161873eb4a68.yaml
Comment (1ucian0) [inline releasenotes/notes/0.18/lo-freqs-circuit-c2ec161873eb4a68.yaml:57]:
```suggestion
    new kwargs, ``qubit_lo_range`` and ``meas_lo_range``, on its constructor.
```

File: releasenotes/notes/0.18/prepare-0.18-9a0fb87005e314fd.yaml
Comment (1ucian0) [inline releasenotes/notes/0.18/prepare-0.18-9a0fb87005e314fd.yaml:19]:
```suggestion
    :mod:`qiskit.circuit.qpy_serialization` module. For example, to generate a
```

File: releasenotes/notes/0.18/prepare-0.18-9a0fb87005e314fd.yaml
Comment (1ucian0) [inline releasenotes/notes/0.18/prepare-0.18-9a0fb87005e314fd.yaml:18]:
```suggestion
    serialization tooling is available  via the
```

File: releasenotes/notes/0.18/set_config-c96a56d1b860386c.yaml
Comment (1ucian0) [inline releasenotes/notes/0.18/set_config-c96a56d1b860386c.yaml:14]:
```suggestion
    If no ``file_path`` argument is specified, the currently used path to the
```

**Review by 1ucian0 — COMMENTED** [review]
Some spelling check

File: releasenotes/notes/0.18/latex_style_user_math_text-266f9b411317ae6b.yaml
Comment (mtreinish) [reply releasenotes/notes/0.18/latex_style_user_math_text-266f9b411317ae6b.yaml:20]: This actually should be

```suggestion
    change a gate color:
```

two `:` will be interpreted as a code block which will undo the jupyter execute directive

File: releasenotes/notes/0.18/add-alignment-management-passes-650b8172e1426a73.yaml
Comment (kdk) [inline releasenotes/notes/0.18/add-alignment-management-passes-650b8172e1426a73.yaml:16]:
```suggestion
    given the constraints of the backend. The
```

File: releasenotes/notes/0.18/add-alignment-management-passes-650b8172e1426a73.yaml
Comment (kdk) [inline releasenotes/notes/0.18/add-alignment-management-passes-650b8172e1426a73.yaml:22]:
```suggestion
    a :class:`~qiskit.circuit.QuantumCircuit` object) are valid given
    an alignment constraint for the target backend.
```

File: releasenotes/notes/0.18/add-initialize-from-method-to-schedules-ce86083b5bd39e6b.yaml
Comment (kdk) [inline releasenotes/notes/0.18/add-initialize-from-method-to-schedules-ce86083b5bd39e6b.yaml:5]:
```suggestion
    :meth:`~qiskit.pulse.Schedule.initialize_from` was added to the
```

File: releasenotes/notes/0.18/add-initialize-from-method-to-schedules-ce86083b5bd39e6b.yaml
Comment (kdk) [inline releasenotes/notes/0.18/add-initialize-from-method-to-schedules-ce86083b5bd39e6b.yaml:8]: Do we know which attributes are carried over? `name` is in the example, but is it all attributes on the `Schedule`?

File: releasenotes/notes/0.18/add-pauli-list-5644d695f91de808.yaml
Comment (kdk) [inline releasenotes/notes/0.18/add-pauli-list-5644d695f91de808.yaml:6]:
```suggestion
    efficiently represent a list of :class:`~qiskit.quantum_info.Pauli`
```

File: releasenotes/notes/0.18/add-scipy-optimizer-d1b6e9b9dcb20356.yaml
Comment (kdk) [inline releasenotes/notes/0.18/add-scipy-optimizer-d1b6e9b9dcb20356.yaml:9]:
```suggestion
    which enables the use of all optimization solvers and all
```

File: releasenotes/notes/0.18/add-scipy-optimizer-d1b6e9b9dcb20356.yaml
Comment (kdk) [inline releasenotes/notes/0.18/add-scipy-optimizer-d1b6e9b9dcb20356.yaml:29]:
```suggestion
    optimizer class are deprecated and will be removed in a future release. Instead, please
```

File: releasenotes/notes/0.18/bugfix-gate-inverse-name-3401093a18813ace.yaml
Comment (kdk) [inline releasenotes/notes/0.18/bugfix-gate-inverse-name-3401093a18813ace.yaml:6]:
```suggestion
    Fixed an issue in the assignment of the :attr:`~qiskit.circuit.Gate.name`
    attribute to :class:`~qiskit.circuit.Gate` generated by multiple calls to
    the :meth:`~qiskit.circuit.Gate.inverse`` method. Prior to this fix
```

File: releasenotes/notes/0.18/add-initialize-from-method-to-schedules-ce86083b5bd39e6b.yaml
Comment (mtreinish) [reply releasenotes/notes/0.18/add-initialize-from-method-to-schedules-ce86083b5bd39e6b.yaml:8]: It looks like it's just `name` and `metadata`:  https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/pulse/schedule.py#L170-L178 and for schedule block it also adds `alignment_context`: https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/pulse/schedule.py#L963-L977

File: releasenotes/notes/0.18/TensoredMeasFitter-d327f0e2d736efdd.yaml
Comment (Cryoris) [inline releasenotes/notes/0.18/TensoredMeasFitter-d327f0e2d736efdd.yaml:10]: I think it's optional and per default does measurement mitigation on each qubit individually
```suggestion
    you will also be able to set the new ``mit_pattern`` kwarg to specify the
```

File: releasenotes/notes/0.18/remove-node-condition-d46579359703a027.yaml
Comment (kdk) [inline releasenotes/notes/0.18/remove-node-condition-d46579359703a027.yaml:7]:
```suggestion
    of the :class:`~qiskit.dagcircuit.DAGDepNode` have been deprecated and
```

File: releasenotes/notes/0.18/set_config-c96a56d1b860386c.yaml
Comment (kdk) [inline releasenotes/notes/0.18/set_config-c96a56d1b860386c.yaml:6]:
```suggestion
    values in a user config from the Qiskit API. For example::
```

File: releasenotes/notes/0.18/support-gradient-for-lbfgsb-87a7261ca79c3ef0.yaml
Comment (kdk) [inline releasenotes/notes/0.18/support-gradient-for-lbfgsb-87a7261ca79c3ef0.yaml:7]:
```suggestion
    gradients are used, the gradient was incorrectly passed as a numpy array
```

File: releasenotes/notes/0.18/lo-freqs-circuit-c2ec161873eb4a68.yaml
Comment (mtreinish) [reply releasenotes/notes/0.18/lo-freqs-circuit-c2ec161873eb4a68.yaml:2]: Should I just delete the release note file? I'm fine with doing this, I tried my best to improve it but the feature itself is confusing to explain.

File: releasenotes/notes/0.18/add-alignment-management-passes-650b8172e1426a73.yaml
Comment (kdk) [inline releasenotes/notes/0.18/add-alignment-management-passes-650b8172e1426a73.yaml:15]: Was this due to #6705 ?

Comment (mtreinish) [reply releasenotes/notes/0.18/add-alignment-management-passes-650b8172e1426a73.yaml:16]: It was, but it was a mistake on my part. I've fixed it in https://github.com/Qiskit/qiskit-terra/pull/6652/commits/6155b6ad7f4e12f43f27733a4526739b293a35ba

**Review by kdk — APPROVED** [review]

**Comment by mtreinish** [comment]
ok, now that #6581 has merged this is unblocked now and we can release. Once this is approved and merged we'll tag the 0.18.0 release

**Review by kdk — APPROVED** [review]

**Pull request merged**
